### DISCUSSION TITLE
开源协议问题

### DISCUSSION THREAD

**Discussion by Anillc** [mamoe/mirai#893] [Q&A] [answered]
假设我开发了一个以MIT协议开源的机器人框架并实现了一个mirai的适配器，适配器可以由用户自己选择，框架本身不依赖于mirai，请问这种情况下框架是否需要以AGPL开源？

**Comment by PeratX** [comment, marked as answer]
可以不使用AGPLv3开源，但是适配器需要使用AGPLv3开源。但是mirai官方推荐所有与mirai相关的项目均采用AGPLv3协议开源。
如果该项目（指框架）不采用AGPLv3协议开源，其极大概率不会被mirai官方推荐或在任何mirai官方页面出现。

**Reply by Anillc** [reply]
那么是否需要另开repo或者说在适配器目录下放AGPL的证书是否规范呢

**Reply by Anillc** [reply]
蒜了蒜了，就用AGPL叭w
